src/useQueryParam.ts

```diff
@@ -44,35 +44,36 @@ export const useQueryParam = <D, D2 = D>(
     refLocation.current = location;
   }, [location]);
 
+  let parsed: EncodedQueryWithNulls;
   // read in the raw query
-  if (!rawQuery) {
+  if (rawQuery) {
+    parsed = rawQuery;
+  } else {
     const locationIsObject = typeof location === 'object';
     const windowIsDefined = typeof window !== 'undefined';
-    rawQuery = React.useMemo(() => {
-      let pathname = {};
-
+    parsed = React.useMemo(() => {
       // handle checking SSR (#13)
       if (locationIsObject) {
         // in browser
         if (windowIsDefined) {
-          pathname = parseQueryString(location.search);
+          return parseQueryString(location.search);
         } else {
           // not in browser
           let url = location.pathname;
           if (location.search) {
             url += location.search;
```

Comment:
Unrelated to my PR, but I wanted to double-check:

The entire `else` block seems to be contingent on `pathname` containing bits of the "query" in certain environments?

Because it is also appending `location.search`, which is already handled by the previous case (the main `if` block)

Reply:
tbh some random person wrote that SSR check to get their next.js stuff working and I didn't care to really look deeply at it or even verify it worked as I don't use it next.js/SSR. No idea what their debugging process was but apparently this worked for them. it seems bizarre to me but I'm not gonna touch it until I feel like running it myself and verifying